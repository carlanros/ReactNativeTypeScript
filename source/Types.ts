import { NavigatorTypes } from './navigator/';
import { LoginTypes } from './login/';
import { UtilityTypes } from './utility/';
import { StorageTypes } from './storage';

/**
 * The global application state as contained in Redux
 *
 * All state objects might not be available.
 *
 * @export
 * @interface appTypes.IApplicationState
 */
export interface IApplicationState {
  route?: NavigatorTypes.INavigationState;
  login?: LoginTypes.ILoginState;
  utility?: UtilityTypes.IUtilityState;
  storage?: StorageTypes.IStorageState;
}

/**
 * Describes possible props for a components.
 *
 * All actions and state objects might not be available.
 *
 * @export
 * @interface IProps
 */
export interface IProps extends IApplicationState {
  // AppLoadingChanged?: appActions.AppLoadingChanged;
  dispatch?: any;
  FacebookLogin?: LoginTypes.FacebookLogin;
  AppErrorChanged?: UtilityTypes.IAppErrorChanged;
  AppLoadingChanged?: UtilityTypes.IAppLoadingChanged;
}
