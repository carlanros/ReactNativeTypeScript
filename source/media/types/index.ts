// media/types/index.ts

export enum MediaConstants {
  /**
   * Login failed or was cancelled
   */
  MEDIA_FAILED = 'MEDIA_FAILED',
  /**
   * Login state: authentication succeeded
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_SUCCESS = 'MEDIA_SUCCESS',

  /**
   * Login state: authentication in progress
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_BUSY = 'Kollar vem du Ã¤r',
}

export interface IMediaState {
  cancelled: boolean;
  height: number;
  width: number;
  uri: string;

}
export interface IMediaAction extends IMediaState {
  type: MediaConstants;
}
