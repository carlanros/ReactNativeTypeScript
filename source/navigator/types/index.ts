export interface INavigationState {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}
