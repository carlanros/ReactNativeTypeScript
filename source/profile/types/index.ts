// profile/types/index
import * as types from '../../Types';

export interface IProfileProps extends types.IProps {
  Logout?: () => any;
}
