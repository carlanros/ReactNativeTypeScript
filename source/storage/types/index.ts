// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
    STORAGE_GOTKEY = 'STORAGE = GOT KEY',
    STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
    STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
    STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
    STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
    STORAGE_SAVED = 'STORAGE = SAVED KEY',
    STORAGE_USER_KEY = 'STORAGE KEY = USER',
    STORAGE_BUSY = 'Friskar upp minnet',
  }

export interface IKey {
/**
 * The key in app storage
 *
 * @type {string}
 */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {

  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
}

export interface IStorageAction extends IStorageState {
  type: StorageConstants;
}