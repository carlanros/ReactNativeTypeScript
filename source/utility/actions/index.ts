// utility/actions/index
import { AppErrorChanged, AppLoadingChanged } from './AppActions';

export { AppErrorChanged, AppLoadingChanged };
