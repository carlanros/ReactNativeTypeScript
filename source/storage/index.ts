// login/index
import StorageReducer from './reducers';
import * as StorageTypes from './types';
import { SaveByKey, GetByKey, RemoveKey } from './actions';

export { StorageReducer, StorageTypes, SaveByKey, GetByKey, RemoveKey  };
