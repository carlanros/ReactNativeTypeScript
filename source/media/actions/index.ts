// media/actions/intex.ts
import * as Redux from 'redux';
import axios from 'axios';
import { Constants, Facebook } from 'expo';
import * as mediaTypes from '../types/';
import { AppErrorChanged, AppLoadingChanged } from '../../utility/';
import { StorageTypes, SaveByKey } from '../../storage';
// import { MediaConstants } from '../types/index';

const takePhoto = 'TakePhoto';
/**
 * Login to Facebook and store user to AsyncStorage
 * @function FacebookLogin
 * @returns {UserDispatch}
 * @memberof FacebookActions
 * @public
 * @global
 */
export const FacebookLogin = () => async (
  dispatch: Redux.Dispatch<mediaTypes.IMediaAction>
) => {
  dispatch(
    AppLoadingChanged({
      isBusy: true,
      reason: mediaTypes.MediaConstants.MEDIA_BUSY,
      sender: takePhoto,
    })
  );

  try {
// Expo bild:

      let actionResult: mediaTypes.IMediaAction = {
          type: mediaTypes.MediaConstants.MEDIA_SUCCESS,

      };

      if (!actionResult.cancelled) {
      dispatch(actionResult);
         // Reset errors
   dispatch(
       AppErrorChanged({ hasError: false, sender: takePhoto }));
    } else {

    dispatch(
        AppErrorChanged({
          hasError: true,
          reason: 'Photo was cancelled',
          exception: mediaTypes.MediaConstants.MEDIA_FAILED,
          sender: takePhoto,
        })
      );
    }
     } catch (error) {
    dispatch(
      AppErrorChanged({
        hasError: true,
        reason: mediaTypes.MediaConstants.MEDIA_FAILED,
        exception: error,
        sender: takePhoto,
      })
    );
  } finally {
    dispatch(AppLoadingChanged({ isBusy: false, sender: takePhoto }));

  }
};