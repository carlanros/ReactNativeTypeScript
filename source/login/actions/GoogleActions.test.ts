// login/actions/index.test
// @ts-ignore
/// <reference types="jest"/>
import configureMockStore from 'redux-mock-store';
import AxiosMockFactory from 'axios-mock-adapter';

import { Store } from '../../Store';
import * as loginTypes from '../types';
import * as types from '../../Types';
import { GoogleLogin } from '../actions/GoogleActions';
import { UtilityTypes } from '../../utility/';
import { StorageConstants } from '../../storage/types';

/**
 * Mock Redux store
 */
const store = configureMockStore<Store>();

// Navigator will call connected components
jest.mock('../../navigator', () => ({
  Routes: 'View',
  AppNavigator: 'Button',
}));

function NoError(
  sender: string,
  reason: string = null,
  exception: string = null
) {
  return {
    sender,
    type: UtilityTypes.UtilityConstants.APP_NO_ERROR,
    reason,
    exception,
  };
}

function AppLoading(sender: string, reason: string) {
  return {
    sender,
    reason,
    type: UtilityTypes.UtilityConstants.APP_LOAD_BUSY,
  };
}

function AppLoadDone(sender: string, reason: string = null) {
  return {
    sender,
    type: UtilityTypes.UtilityConstants.APP_LOAD_DONE,
    reason,
  };
}

describe('Google actions', () => {
  beforeEach(() => {
    // @ts-ignore
    store.clearActions();
  });

  test('GoogleLogin returns SUCCESS when done', async () => {
    const expectedPayload = {
      name: 'test',
      picture: 'http://www.crossplatform,se/',
      accessToken: 'id',
      type: 'Google',
    };

    const expectedActions = [
      AppLoading('GoogleLogin', loginTypes.LoginConstants.LOGIN_BUSY),
      {
        type: loginTypes.LoginConstants.LOGIN_SUCCESS,
        payload: expectedPayload,
      },
      AppLoading('SaveByKey', StorageConstants.STORAGE_BUSY),
      {
        type: StorageConstants.STORAGE_SAVED,
        key: StorageConstants.STORAGE_USER_KEY,
        value: expectedPayload,
      },
      NoError('SaveByKey'),
      AppLoadDone('SaveByKey'),
      NoError('GoogleLogin'),
      AppLoadDone('GoogleLogin'),
    ];

    // Dispatch action
    // @ts-ignore
    await store.dispatch(GoogleLogin());

    // @ts-ignore
    expect(store.getActions()).toMatchSnapshot();
    // @ts-ignore
    expect(store.getActions()).toEqual(expectedActions);
  });
});
