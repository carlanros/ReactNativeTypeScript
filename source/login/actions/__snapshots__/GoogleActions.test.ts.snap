// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google actions GoogleLogin returns SUCCESS when done 1`] = `
Array [
  Object {
    "reason": "Kollar vem du Ã¤r",
    "sender": "GoogleLogin",
    "type": "APP = BUSY LOADING",
  },
  Object {
    "payload": User {
      "accessToken": "id",
      "name": "test",
      "picture": "http://www.crossplatform,se/",
      "type": "Google",
    },
    "type": "LOGIN_SUCCESS",
  },
  Object {
    "reason": "Friskar upp minnet",
    "sender": "SaveByKey",
    "type": "APP = BUSY LOADING",
  },
  Object {
    "key": "STORAGE KEY = USER",
    "type": "STORAGE = SAVED KEY",
    "value": User {
      "accessToken": "id",
      "name": "test",
      "picture": "http://www.crossplatform,se/",
      "type": "Google",
    },
  },
  Object {
    "exception": null,
    "reason": null,
    "sender": "SaveByKey",
    "type": "APP NO ERROR",
  },
  Object {
    "reason": null,
    "sender": "SaveByKey",
    "type": "APP = DONE LOADING",
  },
  Object {
    "exception": null,
    "reason": null,
    "sender": "GoogleLogin",
    "type": "APP NO ERROR",
  },
  Object {
    "reason": null,
    "sender": "GoogleLogin",
    "type": "APP = DONE LOADING",
  },
]
`;
